### PR TITLE
OCPBUGS-38101: Bump k8s.io/* packages to v0.30.3

### PR DESCRIPTION
**Pull request by Miciah**
Bump the vendored k8s.io/* packages to v0.30.3.

    go get k8s.io/api@v0.30.3 k8s.io/apimachinery@v0.30.3 k8s.io/apiserver@v0.30.3 k8s.io/client-go@v0.30.3
    go mod tidy
    go mod vendor

### CODE REVIEW

**Comment by frobware** [comment]
error: build error: creating build container: initializing source docker://image-registry.openshift-image-registry.svc:5000/ci-op-2rgrmfyq/pipeline@sha256:45b675e920b90a914a84b6a5d9e997309b7f608a86eb39a2689b561fb58ed5f0: unable to retrieve auth token: invalid username/password: unauthorized: unable to validate token

/test verify

**Comment by frobware** [comment]
- unit - error: build error: creating build container: initializing source docker://image-registry.openshift-image-registry.svc:5000/ci-op-2rgrmfyq/pipeline@sha256:45b675e920b90a914a84b6a5d9e997309b7f608a86eb39a2689b561fb58ed5f0: unable to retrieve auth token: invalid username/password: unauthorized: unable to validate token

- images - error: build error: creating build container: initializing source docker://image-registry.openshift-image-registry.svc:5000/ci-op-2rgrmfyq/pipeline@sha256:45b675e920b90a914a84b6a5d9e997309b7f608a86eb39a2689b561fb58ed5f0: unable to retrieve auth token: invalid username/password: unauthorized: unable to validate token

- e2e-agnostic - ERRO[2024-08-07T14:15:18Z] could not run steps: step src failed: error occurred handling build src-amd64: could not get build src-amd64: the server is currently unable to handle the request (get builds.build.openshift.io src-amd64) 

Job failures appear to be infra issues based on a sampling of the failures. 

/lgtm
/approve

**Comment by frobware** [comment]
- error: error creating buildah builder: initializing source docker://registry.build02.ci.openshift.org/ci/managed-clonerefs@sha256:cd35f2e1a9137f8f1fc86dbb0e2aa3e96344a064f26473b32359e5b887e9a3e4: reading manifest sha256:cd35f2e1a9137f8f1fc86dbb0e2aa3e96344a064f26473b32359e5b887e9a3e4 in registry.build02.ci.openshift.org/ci/managed-clonerefs: unknown: unable to pull manifest from quay-proxy.ci.openshift.org/openshift/ci@sha256:cd35f2e1a9137f8f1fc86dbb0e2aa3e96344a064f26473b32359e5b887e9a3e4: manifest unknown: manifest unknown


/retest-required

File: go.mod [hunk 1]
```diff
@@ -100,16 +101,16 @@ require (
...
 	gopkg.in/yaml.v3 v3.0.1 // indirect
-	k8s.io/component-base v0.29.1 // indirect
-	k8s.io/kms v0.29.1 // indirect
-	k8s.io/kube-openapi v0.0.0-20231010175941-2dd684a91f00 // indirect
+	k8s.io/component-base v0.30.3 // indirect
+	k8s.io/kms v0.30.3 // indirect
+	k8s.io/kube-openapi v0.0.0-20240228011516-70dd3763d340 // indirect
 	k8s.io/utils v0.0.0-20230726121419-3b25d923346b // indirect
-	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.28.0 // indirect
+	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.29.0 // indirect
```
Comment (candita) [inline go.mod:113]: Is there a v0.30.0 we'd want for apiserver-network-proxy/konnectivity-client?

Comment (Miciah) [reply go.mod:113]: There is a v0.30.0: https://github.com/kubernetes-sigs/apiserver-network-proxy/releases/tag/v0.30.0

There is even a v0.30.3: https://github.com/kubernetes-sigs/apiserver-network-proxy/releases/tag/v0.30.3

However, `sigs.k8s.io/apiserver-network-proxy/konnectivity-client` is an indirect dependency, and the only thing we use it for is to get some default retry-backoff parameters:

```text
% go mod why -m sigs.k8s.io/apiserver-network-proxy/konnectivity-client
# sigs.k8s.io/apiserver-network-proxy/konnectivity-client
github.com/openshift/router/pkg/cmd/infra/router
k8s.io/apiserver/pkg/server/options
k8s.io/apiserver/pkg/server/egressselector
sigs.k8s.io/apiserver-network-proxy/konnectivity-client/pkg/client
% 
```
https://github.com/openshift/router/blob/1c32ae6c0a8594e5a5882a3ebe37370d7513fc04/pkg/cmd/infra/router/template.go#L29
https://github.com/openshift/router/blob/1c32ae6c0a8594e5a5882a3ebe37370d7513fc04/pkg/cmd/infra/router/template.go#L655
https://github.com/openshift/router/blob/1c32ae6c0a8594e5a5882a3ebe37370d7513fc04/vendor/k8s.io/apiserver/pkg/server/options/authentication.go#L41-L50

Now, the package does have some `init` funcs:
https://github.com/openshift/router/blob/1c32ae6c0a8594e5a5882a3ebe37370d7513fc04/vendor/k8s.io/apiserver/pkg/server/options/admission.go#L51-L55
https://github.com/openshift/router/blob/1c32ae6c0a8594e5a5882a3ebe37370d7513fc04/vendor/k8s.io/apiserver/pkg/server/options/tracing.go#L51-L53
As these `init` funcs are just initializing schemata that we don't use, they should be harmless.  

Anyway, the changes between v0.29.0 and v0.30.3 do not include any to `konnectivity-client/pkg/client`: https://github.com/kubernetes-sigs/apiserver-network-proxy/compare/v0.29.0...v0.30.3

So bumping the package would not change any code that we actually use from the package.

**Pull request merged**